### PR TITLE
Typeface simulation fixes

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

This PR ensures the system font collection tries to create a synthetic typeface if no match is found. This matches what the embedded font collection did in the past.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

### CODE REVIEW

File: src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ namespace Avalonia.Media.Fonts
     public abstract class FontCollectionBase : IFontCollection
     {
         protected readonly ConcurrentDictionary<string, ConcurrentDictionary<FontCollectionKey, IGlyphTypeface?>> _glyphTypefaceCache = new();
+        protected IFontManagerImpl? _fontManager;
```
Comment (MrJul) [inline src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs:15]: I don't think we should expose such a field publicly (same for `glyphTypefaceCache` above, but that ship has sailed - it should probably be removed in v12).

Instead, `TryCreateSynthethicGlyphTypeface()` could be static and take an extra `IFontManagerImpl` parameter. Each of SystemFontCollection can keep its private existing field.

**Review by MrJul — COMMENTED** [review]
By reading the code I assume that this PR now applies font simulations to embedded fonts? Should we add some render test to ensure we don't regress this?

File: src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs [hunk 2]
```diff
@@ -80,6 +81,47 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
 
+        protected bool TryCreateSyntheticGlyphTypeface(IGlyphTypeface glyphTypeface, FontStyle style, FontWeight weight,
```
Comment (MrJul) [inline src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs:84]: Should probably be `private protected` so it isn't part of the public API.

Comment (Gillibald) [reply src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs]: I have moved that method to FontManger and made it internal

**Comment by Gillibald** [comment]
This should no longer have any public API changes

**Review by MrJul — APPROVED** [review]

**Pull request merged**